### PR TITLE
🗑️ Remove filter sidebar completely

### PR DESCRIPTION
**Pull request by iethree**
Screen Shot 2022-07-20 at 2 46 52 PM

## Stuff Removed
- removed filter sidebar component
- removed redux actions and selectors for filter sidebar
- removed conditionals from filter components that had different styling for sidebar and popover variants
- removed filter sidebar specific tests

## Testing Steps
- other sidebars (like the summarize sidebar) continue to work
- filter modal still works
- filters still work in notebook mode
- filters in column header popovers still work

### CODE REVIEW

File: frontend/test/metabase/scenarios/filters/filter.cy.spec.js [hunk 1]
```diff
@@ -695,7 +673,7 @@ describe("scenarios > question > filter", () => {
     cy.findByText("Tax").should("not.exist");
   });
 
-  it.skip("should work on twice summarized questions (metabase#15620)", () => {
+  it("should work on twice summarized questions (metabase#15620)", () => {
```
Comment (iethree) [inline frontend/test/metabase/scenarios/filters/filter.cy.spec.js:676]: I fixed this test to work with the bulk filter modal

File: frontend/src/metabase/query_builder/components/filters/FilterPopover.tsx [hunk 2]
```diff
@@ -48,7 +48,6 @@ type Props = {
   dateShortcutOptions?: DateShortcutOptions;
   showCustom?: boolean;
   isNew?: boolean;
-  isSidebar?: boolean;
```
Comment (iethree) [inline frontend/src/metabase/query_builder/components/filters/FilterPopover.tsx:51]: The filter sidebar was just a wrapper around the filter popover with a bunch of props drilled down into various components to trigger layout differences. We can remove all those conditionals and simplify a bunch of these components now.

File: frontend/src/metabase/query_builder/components/filters/pickers/DefaultPicker.jsx [hunk 3]
```diff
@@ -178,7 +176,7 @@ export default function DefaultPicker({
 
   return (
     <DefaultPickerContainer
-      limitHeight={!isSidebar}
+      limitHeight
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/filters/pickers/DefaultPicker.jsx:179]: nit: This is the only place that `DefaultPickerContainer` is used. So, it should be safe to modify their definition to not rely on this `limitHeight` prop anymore

https://github.com/metabase/metabase/blob/6c43ace1d1be0a133fcd487f29f33105af40427f/frontend/src/metabase/query_builder/components/filters/pickers/DefaultPicker.styled.tsx#L26-L34

File: frontend/src/metabase/query_builder/components/filters/pickers/TimePicker/TimePicker.jsx [hunk 4]
```diff
@@ -36,8 +36,8 @@ const SingleTimePicker = ({ filter, onFilterChange }) => (
 
 SingleTimePicker.horizontalLayout = true;
 
-const MultiTimePicker = ({ filter, onFilterChange, isSidebar }) => (
-  <MultiTimePickerRoot direction={isSidebar ? "column" : "row"}>
+const MultiTimePicker = ({ filter, onFilterChange }) => (
+  <MultiTimePickerRoot direction="row">
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/filters/pickers/TimePicker/TimePicker.jsx:40]: I doubt direction used to do anything if you look at its definition

https://github.com/metabase/metabase/blob/25d1a66109de634ade9c9b33844ee42aff097036/frontend/src/metabase/query_builder/components/filters/pickers/TimePicker/TimePicker.styled.tsx#L13-L26

here's when it was introduced.

https://github.com/metabase/metabase/pull/23398/files#diff-e90e927d0004a5e18592c38b369e3f071618e98327e9f497ec92a39687094f66

Seems like `MultiTimePickerRootProps` are mentioned but never used.

It's also not a valid HTML attribute https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes

File: frontend/test/metabase/scenarios/filters/filter.cy.spec.js [hunk 5]
```diff
@@ -200,29 +201,6 @@ describe("scenarios > question > filter", () => {
     cy.findByText(AGGREGATED_FILTER);
   });
 
-  it.skip("in a simple question should display popup for custom expression options (metabase#14341) (metabase#15244)", () => {
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/filters/filter.cy.spec.js:203]: Can this be removed? I know it was skipped, but not sure if this relates to the removed filter sidebar or not.

**Review by WiNloSt — APPROVED** [review]
RIP `<FilterSidebar />` 🌹

File: frontend/test/metabase/scenarios/filters/filter.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/filters/filter.cy.spec.js:676]: Did you fix the test or the issue?
If https://github.com/metabase/metabase/issues/15620 is now working, we should close it.

Comment (iethree) [reply frontend/test/metabase/scenarios/filters/filter.cy.spec.js:676]: oh interesting, I think the issue is still there, but the reproduction doesn't actually reproduce the bug accurately. I'll fix the repro

File: frontend/test/metabase/scenarios/filters/filter.cy.spec.js (hunk 5 above)
Comment (iethree) [reply frontend/test/metabase/scenarios/filters/filter.cy.spec.js:203]: unlike the sidebar, the modal isn't supposed to support custom expressions.

File: frontend/test/metabase/scenarios/filters/filter.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/filters/filter.cy.spec.js:676]: What is inaccurate in the repro (apart from the fact that the sidebar doesn't exist anymore)?

Comment (iethree) [reply frontend/test/metabase/scenarios/filters/filter.cy.spec.js:676]: Currently, the repro correctly applies a filter, and draws a chart (which is what the assertion is checking) but removes one of the summaries in the process.

Screen Shot 2022-07-21 at 9 59 34 AM

**Pull request merged**